src/Twig/Handler/ArrayHandler.php

```diff
@@ -110,8 +110,8 @@ private function orderHelper($a, $b)
                 return 0;
             }
 
-            $aVal = $a[$this->order_on_secondary];
-            $bVal = $b[$this->order_on_secondary];
+            $aVal = (is_array($a) ? $a[$this->order_on] : $a->{$this->order_on_secondary});
+            $bVal = (is_array($b) ? $b[$this->order_on] : $b->{$this->order_on_secondary});
```

Comment:
Also the outer `()` isn't needed… but that is just OOE OCD :stuck_out_tongue_winking_eye: